File: solutions/1-two-sum/two-sum.rb
def two_sum(nums, target)
    ht = {}
    
    nums.each_with_index do |num, i|
        if ht.key?(target - num)
            return [ht[target - num], i]
        end
        
        ht[num] = i
    end
    
    []
end

File: solutions/1-two-sum/two-sum.js
// JavaScript


var twoSum = function(nums, target) {
    const ht = new Map();
    
    for (let i = 0; i < nums.length; i++) {
        const num = nums[i];
        
        if (ht.has(target - num)) {
            return [ht.get(target - num), i];
        }
        
        ht.set(num, i);
    }
    
    return [];
};